### PR TITLE
start using gcp-metadata for metadata queries

### PR DESCRIPTION
**Pull request by ofrobots**

### CODE REVIEW

File: src/agent/debuglet.js [hunk 1]
```diff
@@ -263,30 +264,31 @@ Debuglet.createDebuggee =
...
-    Debuglet.prototype.getProjectId_ = function(callback) {
+/**
+ * @private
+ */
+Debuglet.prototype.getProjectId_ = function(callback) {
   var that = this;
 
   // We need to figure out whether we are running on GCP. We can use our ability
   // to access the metadata service as a test for that.
   // TODO: change this to getProjectId in the future.
```
Comment (matthewloring) [inline src/agent/debuglet.js]: Can this TODO go away now?

File: src/agent/debuglet.js [hunk 2]
```diff
@@ -263,30 +264,31 @@ Debuglet.createDebuggee =
...
-                  metadataProject;
-
-    // We if don't have a projectId by now, we fail with an error.
-    if (!project) {
-      return callback(err);
-    }
-    return callback(null, project, onGCP);
-  });
+  metadata.project(
+      'numeric-project-id', function(err, response, metadataProject) {
```
Comment (matthewloring) [inline src/agent/debuglet.js]: Could we make this key available as a constant.

File: src/agent/debuglet.js [hunk 3]
```diff
@@ -263,30 +264,31 @@ Debuglet.createDebuggee =
...
-    var project = that.debug_.options.projectId || process.env.GCLOUD_PROJECT ||
-                  metadataProject;
-
-    // We if don't have a projectId by now, we fail with an error.
-    if (!project) {
-      return callback(err);
-    }
-    return callback(null, project, onGCP);
-  });
+  metadata.project(
```
Comment (matthewloring) [inline src/agent/debuglet.js]: Does this have different retry behavior than what existed before?

File: test/standalone/test-debuglet.js [hunk 4]
```diff
@@ -48,6 +48,7 @@ describe(__filename, function() {
   });
 
   it('should not start when projectId is not available', function(done) {
+    this.timeout(8000);
```
Comment (matthewloring) [inline test/standalone/test-debuglet.js]: Yikes, is the new query approach slower?

File: src/agent/debuglet.js (hunk 1 above)
Comment (ofrobots) [inline src/agent/debuglet.js]: In a follow-on change. Sure.

File: src/agent/debuglet.js (hunk 2 above)
Comment (ofrobots) [inline src/agent/debuglet.js]: PRs on `gcp-metadata` are welcome!

File: test/standalone/test-debuglet.js (hunk 4 above)
Comment (ofrobots) [inline test/standalone/test-debuglet.js]: I think we should optionally disable retry in `gcp-metadata`, but I don't think it is required for this PR.

File: src/agent/debuglet.js (hunk 3 above)
Comment (ofrobots) [inline src/agent/debuglet.js]: Yes.

File: src/agent/debuglet.js (hunk 2 above)
Comment (matthewloring) [inline src/agent/debuglet.js]: My thought was to expose it as a constant within our own module but if it's only ever referenced here that's fine

**Pull request merged**